openlcsd/flow/tasks.py

```diff
@@ -975,6 +975,13 @@ def sync_result_to_corgi(context, engine):
     if not source:
         # Component without source info means no scan is done.
         # FIXME: are there any other circumstances cause no source?
+        engine.logger.warning(f"Component({component_uuid}) sync to corgi "
+                              f"failed, no source found.")
+        client.patch(
+            f"components/{olcs_component['id']}",
```

Comment:
If the `source` is not found, I think maybe the `id` is also not found.

Reply:
I've seen cases where component exists but source does not.

Reply:
OK